app/src/main/kotlin/com/hippo/ehviewer/coil/LimitConcurrencyNetworkClient.kt

```diff
@@ -0,0 +1,27 @@
+package com.hippo.ehviewer.coil
+
+import coil3.network.NetworkClient
+import coil3.network.NetworkRequest
+import coil3.network.NetworkResponse
+import coil3.network.ktor3.asNetworkClient
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.URL_PREFIX_THUMB_E
+import com.hippo.ehviewer.client.URL_PREFIX_THUMB_EX
+import io.ktor.client.HttpClient
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withPermit
+
+class LimitConcurrencyNetworkClient(val impl: NetworkClient) : NetworkClient {
+    val eSemaphore = Semaphore(Settings.thumbConcurrency)
+    val exSemaphore = Semaphore(Settings.thumbConcurrency)
```

Comment:
We are not gonna requesting e and ex at the same time, I think they can share the semaphore.

Reply:
Done, Actually ehgt don't need semaphore protect.